inference-engine/src/mkldnn_plugin/nodes/mkldnn_bucketize_node.cpp

```diff
@@ -0,0 +1,218 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "base.hpp"
+
+#include <string>
+#include <vector>
+#include <algorithm>
+
+#include <ngraph/opsets/opset3.hpp>
+#include "ie_parallel.hpp"
+#include "mkldnn_bucketize_node.h"
+
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+bool MKLDNNBucketizeNode::isSupportedOperation(const std::shared_ptr<ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        const auto bucketsize = std::dynamic_pointer_cast<const ngraph::opset3::Bucketize>(op);
+        if (!bucketsize) {
+            errorMessage = "Only opset3 Bucketize operation is supported";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+MKLDNNBucketizeNode::MKLDNNBucketizeNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng,
+                                     MKLDNNWeightsSharing::Ptr &cache) : MKLDNNNode(op, eng, cache) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        IE_THROW(NotImplemented) << errorMessage;
+    }
+
+    errorPrefix = "Bucketize layer with name '" + op->get_friendly_name() + "' ";
+    const auto bucketsize = std::dynamic_pointer_cast<const ngraph::opset3::Bucketize>(op);
+
+    if (op->get_input_size() != 2 || op->get_output_size() != 1) {
+        IE_THROW() << errorPrefix << " has incorrect number of input/output edges!";
+    }
+
+    // check one attribute
+    with_right = bucketsize->get_with_right_bound();
+
+    // check precisions for input and output tensors
+    input_precision = getOriginalInputPrecisionAtPort(INPUT_TENSOR_PORT);
```

Comment:
We have to work with precisions not inside the ctor since them might be changed after its call: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp#L283. Please move all precision related code (in this file lines 49:62) into initSupportedPrimitiveDescriptors().
Please apply for all layers.
cc'ed @mandrono

Reply:
Fixed in 5739d74c2b2b49a3c3c4562493dbf7fae30e7364. Thank you!